### PR TITLE
Implement setAuthor and setURL for MessageBuilder

### PR DESCRIPTION
**Pull request by Exitare**
The scope is to implement missing methods mentioned here #52, to be precise setAuthor and setUrl, which sets the author / url respectively.

Additional, those methods are added to the typescript index.d.t.s file.

### CODE REVIEW

File: lib/builder.js [hunk 1]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
...
+    }
+
+    /**
+     * Set the author of the embedded object
+     * @param {string} author the author
+     * @param {string} iconURL Optional: the icon url
+     * @param {string} url Optional: the url
+     */
+    setAuthor(author, iconURL, url) {
+        this.data.attachments[0].author = author;
```
Comment (jos-b) [inline lib/builder.js:112]: Wrong key name, should be `author_name`:
```suggestion
        this.data.attachments[0].author_name = author;
```

File: lib/builder.js [hunk 2]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
...
+
+        if (iconURL) {
+            this.data.attachments[0].icon_url = iconURL;
```
Comment (jos-b) [inline lib/builder.js:115]: Wrong key name, should be `author_icon`:
```suggestion
            this.data.attachments[0].author_icon = iconURL;
```

File: lib/builder.js [hunk 3]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
...
+        }
+
+        if (url) {
+            this.data.attachments[0].url = url;
```
Comment (jos-b) [inline lib/builder.js:119]: Wrong key name, should be `author_link`:
```suggestion
            this.data.attachments[0].author_link = url;
```

File: lib/builder.js [hunk 4]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
         return this;
     }
 
+    /**
+     * Set the embedded url
+     * @param {string} url The url 
+     */
+    setURL(url) {
+        this.data.attachments[0].url = url;
```
Comment (jos-b) [inline lib/builder.js:101]: Wrong key name, should be `title_link`:
```suggestion
        this.data.attachments[0].title_link = url;
```

File: lib/builder.js [hunk 5]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
...
+        return this;
+    }
+
+    /**
+     * Set the author of the embedded object
+     * @param {string} author the author
+     * @param {string} iconURL Optional: the icon url
+     * @param {string} url Optional: the url
+     */
```
Comment (jos-b) [inline lib/builder.js:110]: The docs here could be nicer:
```suggestion
    /**
     * Set the author of the embed
     * @param {string} author Name of the author
     * @param {string} iconURL Optional: Icon URL of the author
     * @param {string} url Optional: Link to the author
     */
```

File: index.d.ts [hunk 6]
```diff
@@ -98,6 +98,26 @@ declare module "webhook-discord" {
      */
     public setTime(timestamp?: number): MessageBuilder;
 
+    /**
+     * Set the title for the discord hook
+     * @param title the webhooks´ title
+     */
```
Comment (jos-b) [inline index.d.ts:104]: This should use the documentation from the JavaScript file.
```suggestion
    /**
     * Set the title of the embed
     *
     * @param {string} title The title to use
     */
```

File: index.d.ts [hunk 7]
```diff
@@ -98,6 +98,26 @@ declare module "webhook-discord" {
...
+    public setTitle(title: string): MessageBuilder;
+
+    /**
+     * Set the author of the embedded object
+     * @param author the authors name
+     * @param iconURL the icon url
+     * @param url the url
+     */
```
Comment (jos-b) [inline index.d.ts:112]: Should mirror the JavaScript:
```suggestion
    /**
     * Set the author of the embed
     * @param {string} author Name of the author
     * @param {string} iconURL Optional: Icon URL of the author
     * @param {string} url Optional: Link to the author
     */
```

File: index.d.ts [hunk 8]
```diff
@@ -98,6 +98,26 @@ declare module "webhook-discord" {
...
+    public setAuthor(author: string, iconURL?: string, url?: string): MessageBuilder;
+
+    /**
+     * Set the url for the discord hook
+     * @param url the url
```
Comment (jos-b) [inline index.d.ts:117]:
```suggestion
     * Set the URL for the Discord embed
     *
     * @param {string} url The URL to link to from the embed.
```

File: lib/builder.js [hunk 9]
```diff
@@ -93,6 +93,35 @@ class MessageBuilder {
         return this;
     }
 
+    /**
+     * Set the embedded url
+     * @param {string} url The url
```
Comment (jos-b) [inline lib/builder.js:98]: Could use a nicer doc comment:
```suggestion
	 * Set the URL for the Discord embed
     *
     * @param {string} url The URL to link to from the embed.
```

**Comment by Exitare** [comment]
Thanks for the review. I thought i was thorough enough to find the correct members. 
Can you link where the members of a message to discord are specified?

**Comment by Exitare** [comment]
0f18fd4643317ae4b7710372f3d54b883ed3c770, should apply the requested changes.

**Comment by jos-b** [comment]
We don't actually use the Discord webhook format, we use the Slack one, so here: https://api.slack.com/reference/messaging/attachments

**Comment by Exitare** [comment]
Thanks!

**Comment by jos-b** [comment]
Released webhook-discord@3.7.0

**Pull request merged**